### ISSUE TITLE
[Suggestion] Automatically set java.exe to ”High priority”

### ISSUE THREAD

**Issue by Nevrai** [killpowa/Twickt-Launcher#1432]
For some reason, GDLauncher always launches the game (java.exe) in low priority, which is definitely not ideal for performance as it tells your computer to focus less of its CPU and GPU power on the game, when it should be focusing most of its resources on it.

Could this be changed so that GDLauncher always launches java.exe in high priority mode, so that it doesn’t need to be changed manually each time the game is launched? If not that, it should at least launch in normal priority, but I think it’d be great if it could launch it in high priority automatically, whether it’s an option in the settings or just an unchangable default.

**Comment by Eskaan** [comment]
I don't think this is really needed. You don't want other programs to run slower because of Minecraft and if you don't need them you shouldn't have them opened.

You mean java.exe not javaw.exe btw

**Comment by vgrynch** [comment]
Well, this depends on how Minecraft is running. When just play single world, it will be worse because server running together with client on the same process will dedicate most resources to server, and thus client would have bad FPS.

When join to separate server, I confirm, for me, with 1.19.2, 360+ mods in modpack and connecting to the server, following is improved when set higher priority for javaw that runs Minecraft:
- startup time (about 15-20% improvement, measured by lightspeed mod)
- resources loading/syncing when joining the server is much faster, especially noticeable for the "Loading terrain" phase which became really much faster, about 4x times
- UI resource pack reloading is 2x faster

**Comment by Eskaan** [comment]
> measured by lightspeed mod

wdym? If I see this right that is a mod that tries to improve startup performance in general by doing modifications to the game.

How did you even set the program to high priority (on startup) to test that?

As you may have noticed, I did not close this issue. This is still open for anyone to go and implement a setting for it.

**Comment by vgrynch** [comment]
> wdym? If I see this right that is a mod that tries to improve startup performance in general by doing modifications to the game.

@Eskaan, Did I say lightspeed was used to set higher priority? I said it is used to measure. It starts measuring time of loading of mods as soon as mods are loaded by forge. So it shows time from that moment till the startup complete. What I meant is when the priority of the process is higher, mods are loaded quicker, that's all.

To set high priority for the process, you need to launch java or javaw process with higher priority. This is done by various OS ways or some tools prior to launching it. The popular method is to use "start /{priority}", but there are also other ways, like, for example, setting the priority of the process after the launching. You can search for it. One of the good ways is discussed here:
https://www.reddit.com/r/dailyscripts/comments/1wn8d9/batch_start_minecraft_as_realtime_process/
